app/src/main/java/com/android/sample/ui/camera/Gallery.kt

```diff
@@ -53,24 +73,86 @@ fun ProfileImage(userId: String, modifier: Modifier = Modifier, imageViewModel:
         })
   }
 
+  LaunchedEffect(bitmap) {
+    if (bitmap != null && editing) {
+      isImageRemoved = false
+    } else if (bitmap == null && imageUrl == null && editing) {
+      isImageRemoved = true
+    }
+  }
+  LaunchedEffect(imageUrl) {
```

Comment:
the 2 launchedEffect blocks can be combined into 1 here :)
for example: 
isImageRemoved = when {
    bitmap != null && editing -> false
    imageUrl == null && bitmap == null && editing -> true
    else -> isImageRemoved
  }
  In this case, don't forget to give LaunchedEffect the 2 parameters, bitmap & imageUrl

Reply:
Thank you for your suggestion, however here using launched effect is about when to "activate" the clause. One will be activated when the imageUrl will be changed(if the picture is changed, or removed from the storage), and the other will be activated if a picture is put as a replacement of the current profile picture, before saving it. So I think that it is important to keep these 2 blocks as they treat independant cases.

Reply:
Understood! I see that it's more logical if they're independent cases!